### PR TITLE
Update the minimum supported version to WordPress 5.8

### PR DESCRIPTION
**Pull request by youknowriad**
This PR updates the minimum WP version to 5.8 in the plugin. @gziolo suggested that we should change our policy to support only the last WP version at some point. I'd be happy to make that happen separately once it's all discussed properly.

### CODE REVIEW

File: docs/how-to-guides/themes/create-block-theme.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ Description:
 Tags: full-site-editing, blog
 Version: 1.0.0
 Requires at least: 5.0
```
Comment (Mamaduka) [inline docs/how-to-guides/themes/create-block-theme.md]:
```suggestion
Requires at least: 5.9
```

**suggestion (non-blocking):** Maybe we should also update this to 5.9 as well.

**Comment by youknowriad** [comment]
I may have been a bit too aggressive 😬 I removed all widgets/customizer code and all code in lib/wordpress-5.8 folders as well. I probably broke some things in the middle but let's test and fix.

**Review by Mamaduka — COMMENTED** [review]
This looks good. Tested with Classic and Block themes, I've not seen any PHP notices or WSOD.

I think we can also remove these lines:

https://github.com/WordPress/gutenberg/blob/ef22c1636b06ccd4df49e85c4ff56ca18f0ae64a/lib/editor-settings.php#L21-L24

We can also replace `__unstableEnableFullSiteEditingBlocks` with `supportsTemplateMode` on client side.

**Review by noisysocks — APPROVED** [review]
Tested:

- WP 5.9 + Gutenberg
  - ✅ Inserting a post.
  - ✅ Making a change in the site editor with block theme active.
  - ✅ Updating widgets in the widgets editor with classic theme active.
  - ✅ Using the Widget Group block in widgets editor with classic theme active.
  - ✅ Updating widgets in the customizer with classic theme active.
- WP 5.8 + Gutenberg
  - ✅ Inserting a post.
  - ✅ Making a change in the site editor with block theme active.
  - ✅ Updating widgets in the widgets editor with classic theme active.
  - ✅ Using the Widget Group block in widgets editor with classic theme active.
  - ✅ Updating widgets in the customizer with classic theme active.

File: lib/compat/wordpress-5.8.1/index.php [hunk 2]
```diff
@@ -1,9 +0,0 @@
-<?php
-/**
- * Temporary compatibility shims for features present in Gutenberg.
- *
- * @package gutenberg
- */
-
-// Load the polyfills.
-require_once __DIR__ . '/widget-render-api-endpoint/index.php';
```
Comment (noisysocks) [inline lib/compat/wordpress-5.8.1/index.php:9]: Oh, sorry. This ended up landing in WP 5.9 not WP 5.8.1. You'll need to restore the `widget-render-api-endpoint` directory and move it to `lib/compat/wordpress-5.9`.

**Review by noisysocks — CHANGES_REQUESTED** [review]

File: lib/compat/wordpress-5.8.1/index.php (hunk 2 above)
Comment (noisysocks) [reply lib/compat/wordpress-5.8.1/index.php:9]: Done in ae7f42f.

**Review by noisysocks — APPROVED** [review]

**Review by ntsekouras — APPROVED** [review]
LGTM! Thanks Riad!

**Comment by youknowriad** [comment]
Thank you folks for the help landing this.

**Comment by oandregal** [comment]
Hey, the [plugin page](https://wordpress.org/plugins/gutenberg/) still says that it supports 5.7 or higher. Where is that configured?

**Comment by youknowriad** [comment]
@oandregal I think that's because Gutenberg 12.6 is not yet released.

**Comment by oandregal** [comment]
Ah, I see, makes sense.

**Pull request merged**